docs/proposals/cri.md

```diff
@@ -0,0 +1,313 @@
+---
+title: CRI Design for edged
+authors:
+    - "@gpinaik"
+approvers:
+  - "@qizha"
+  - "@CindyXing"
+  - "@Baoqiang-Zhang"
+  - "@m1093782566"
+creation-date: 2019-04-19
+last-updated: 2019-04-21
+status: alpha
+---
+
+# CRI Support in edged
+
+* [CRI Support in edged](#CRI-Support-in-edged)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non\-goals](#non-goals)
+  * [Proposal](#proposal)
+    * [Use Cases](#use-cases)
+  * [High Level Design](#high-level-design)  
+    * [Edged with CRI support](#edged-with-cri-support)
+  * [Low Level Design](#low-level-design)  
+    * [Packages](#packages)
+    * [Configuration parameters](#configuration-parameters)
+    * [Data structure modifications](#data-structure-modifications)
+    * [Edged object creation modifications](#edged-object-creation-modifications)
+    * [Runtime dependent module modifications](#runtime-dependent-module-modifications)
+    * [Runtime dependent functional modifications](#runtime-dependent-functional-modifications)
+  * [Open Questions](#open-questions) 
+    
+    
+## Motivation
+This proposal addresses the Container Runtime Interface support in edged to enable supporting multiple container
+runtimes like docker, containerd, crio etc on the edge platform.
+
+### Goals
+CRI support in edged must:
+* support multiple runtimes like docker, contianerd, crio etc.
+* Support corresponding CNI with pause container and IP
+
+### Non-goals
+
+* Automatic detection of container runtimes and its selection.
+* 
+## Proposal
+
+Currently Kubernetes kubelet CRI supports container runtimes like containerd, crio etc but support for docker runtime is
+provided using dockershim as well. However going forward even docker runtime will be supported through only CRI. Also 
+currently kubeedge edged supports only docker runtime using the legacy dockertools. Hence we propose to support multiple 
+container runtime in kubeedge edged as follows
+1. Include CRI support as in kubernetes kubelet to support contianerd, crio etc
+2. Continue with docker runtime support using legacy dockertools until CRI support for the same is available i.e. support
+for docker runtime using dockershim is not considered in edged
+
+
+### Use Cases
+
+* Customer can choose the required runtime on his edge platform
+
+
+## High Level Design
+
+### Edged with CRI support
+<img src="../docs/images/edged/docker-cri.png">
+
+## Low Level Design
+
+### Packages
+
+Additional CRI runtime packages need to be included.
+
+```go
+"k8s.io/kubernetes/pkg/kubelet/kuberuntime"
+"k8s.io/kubernetes/pkg/kubelet/remote"
+
+```
+
+### Configuration parameters
+
+The options to configure runtime type i.e. docker or other remote runtimes(crio, containerd etc) and the CRI remote endpoints
+for runtime and image services should be provided.
+
+```go
+type Config struct {
```

Comment:
We don't need to show the code-level data structure in design document unless it's API.

Reply:
Specifying the structure as it is used below